recaf-ui/src/main/java/software/coley/recaf/ui/docking/DockingLayoutManager.java

```diff
@@ -0,0 +1,221 @@
+	public static final String ID_SPACE_TOOLS_BOTTOM = "space-tools-bottom";
+
+	private final DockingManager dockingManager;
+	private final Instance<LoggingPane> loggingPaneProvider;
+	private final Instance<WelcomePane> welcomePaneProvider;
+	private final Instance<WorkspaceInformationPane> workspaceInfoProvider;
+	private final Instance<WorkspaceExplorerPane> workspaceExplorerProvider;
+	private final RootDockLayout root;
+
+	@Inject
+	public DockingLayoutManager(@Nonnull DockingManager dockingManager,
+	                            @Nonnull WorkspaceManager workspaceManager,
+	                            @Nonnull Instance<LoggingPane> loggingPaneProvider,
+	                            @Nonnull Instance<WelcomePane> welcomePaneProvider,
+	                            @Nonnull Instance<WorkspaceInformationPane> workspaceInfoProvider,
+	                            @Nonnull Instance<WorkspaceExplorerPane> workspaceExplorerProvider) {
+		this.dockingManager = dockingManager;
+		this.loggingPaneProvider = loggingPaneProvider;
+		this.welcomePaneProvider = welcomePaneProvider;
+		this.workspaceInfoProvider = workspaceInfoProvider;
+		this.workspaceExplorerProvider = workspaceExplorerProvider;
+
+		// Register listener
+		ListenerHost host = new ListenerHost();
+		workspaceManager.addWorkspaceOpenListener(host);
+		workspaceManager.addWorkspaceCloseListener(host);
+
+		// Create root
+		LayoutBuilder builder = dockingManager.getBento().newLayoutBuilder();
+		DockLayout top = newEmptyTop();
+		DockLayout bottom = newBottom();
+		root = builder.root(builder.split(new SplitLayoutArgs()
+				.setOrientation(Orientation.VERTICAL)
+				.setChildrenSizes(-1, 150)
+				.addChildren(top, bottom)));
+	}
+
+	@Nonnull
+	public RootDockLayout getRoot() {
+		return root;
+	}
+
+	@Nonnull
+	private DockSpace newWelcomeSpace() {
+		LayoutBuilder builder = dockingManager.getBento().newLayoutBuilder();
+		return builder.single(new SingleSpaceArgs()
+				.setDockable(builder.dockable()
+						.withNode(welcomePaneProvider.get())
+						.withDragGroup(-1)
+						.build())
+				.setSide(null)
+		);
+	}
+
+	@Nonnull
+	private DockLayout newWorkspaceExplorerLayout() {
+		LayoutBuilder builder = dockingManager.getBento().newLayoutBuilder();
+		return builder.leaf(new LeafLayoutArgs()
+				.setResizeWithParent(false)
+				.setIdentifier(ID_LAYOUT_WORKSPACE_TOOLS)
+				.setSpace(builder.tabbed(
+						new TabbedSpaceArgs()
+								.setIdentifier(ID_SPACE_WORKSPACE_TOOLS)
+								.setCanSplit(false)
+								.addDockables(dockingManager.newToolDockable("workspace.title", CarbonIcons.TREE_VIEW, workspaceExplorerProvider.get()))
+								.setMenuFactory(space -> {
+									ContextMenu menu = new ContextMenu();
+									addSideItems(menu, space);
+									return menu;
+								})
+				))
+		);
+	}
+
+	private static void addSideItems(@Nonnull ContextMenu menu, @Nonnull TabbedDockSpace tabbed) {
+		// TODO: Move this to common location, and make other things use it
```

Comment:
I've not yet figured out where would be best to put this. Mainly since I dont know how many places I want it to be used in. I don't want every tabbed space to have the side being adjustable, so I can't make it the default...